### ISSUE TITLE
TypeError: Cannot read property 'name' of undefined

### ISSUE THREAD

**Issue by hepcat72** [jprichardson/node-fs-extra#926]
- **Operating System:** Raspbian
- **Node.js version:** v14.18.1
- **`fs-extra` version:**:
```
>sudo npm ls -g fs-extra
/usr/local/lib
└─┬ homebridge@1.3.5
  └── fs-extra@10.0.0
```
I hope this is the right place to report this issue.  I see this error in my homebridge installation (which installs fs-extras as a dependency I believe) constantly in my homebridge error log:
```
TypeError: Cannot read property 'name' of undefined
    at exports.fromCallback (/usr/lib/node_modules/homebridge/node_modules/universalify/index.js:15:26)
    at Object.<anonymous> (/usr/lib/node_modules/homebridge/node_modules/fs-extra/lib/fs/index.js:57:27)
    at Module._compile (internal/modules/cjs/loader.js:1085:14)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)
    at Module.load (internal/modules/cjs/loader.js:950:32)
    at Function.Module._load (internal/modules/cjs/loader.js:790:12)
    at Function._load (/usr/lib/node_modules/pm2/node_modules/@pm2/io/src/metrics/httpMetrics.ts:190:35)
    at Module.require (internal/modules/cjs/loader.js:974:19)
    at require (internal/modules/cjs/helpers.js:93:18)
    at Object.<anonymous> (/usr/lib/node_modules/homebridge/node_modules/fs-extra/lib/index.js:5:6)
```
What can I do to address this?

**Comment by RyanZim** [comment]
Are you running a stock version of Node.js. That error indicates that `require('fs').realpath.native` is undefined on your system

**Comment by hepcat72** [comment]
This issue appears to be restricted to when you start homebridge using pm2.  I have worked around the issue by switching to hb-service.

**Comment by roman-teamsystems** [comment]
Hi  @hepcat72.

I have the same issue. Did you solve yours somehow?

**Comment by hepcat72** [comment]
No, I did not, but only because I simply haven't tried to figure it out. I just worked around the issue by switching from pm2 to hb-service to manage my homebridge process, though I am also having issues with node-red running via pm2. I can start it up, but after a reboot, it doesn't run and I have to start it manually. I don't know if it's related to fs* though - and I discovered that I had 2 competing pm2 installations (root vs pi). I uninstalled the root one and crossed my fingers, but I haven't checked with a reboot since then.

But take all this with a huge grain of salt because I don't know what the hell I'm doing. I probably used sudo somewhere where I shouldn't have or something. I'm too busy with a bathroom remodel to figure it out, so I'm just treading water until I have time to set up my new pi 4.
